### PR TITLE
Porting run.sh  (obtained from 0.1.3-SNAPSHOT  )to window batch file 

### PR DESCRIPTION
**Pull request by mingqin1**
Porting run.sh (obtained from 0.1.3-SNAPSHOT )to window batch file run.bat.

Testing environment settings are window 8.1 x64-based processor, JDK 1.6u45  and maven 3.2.1, netbeans 3.7.1
Testing Results:
Just testing batch script to bring up server without adding , deleting, retrieving schema with three back-ends ( in-memory, file-system and zookeeper) , servers were bounced up correctly, except zookeeper 's gave warning messages, which could be caused by missing zookeeper in window testing machine since  CentOS6.4 linux ( missing installation of zookeeper ) had same warning messages

Below are warnings:

WARN  [org.apache.zookeeper.ClientCnxn         ] Session 0x0 for server null, unexpected error, closing socket
connection and attempting reconnect
java.net.ConnectException: Connection refused: no further information
        at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[na:1.6.0_45]
        at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:599) ~[na:1.6.0_45]
        at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361) ~[schema-repo-zk-bundle-0.
1.3-SNAPSHOT-withdeps.jar:0.1.3-SNAPSHOT]
        at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081) ~[schema-repo-zk-bundle-0.1.3-SNAPSHOT-w
ithdeps.jar:0.1.3-SNAPSHOT]

### CODE REVIEW

File: run.bat [hunk 1]
```diff
@@ -0,0 +1,37 @@
+@Echo on
+set JAVA_HOME=C:\Java\jdk1.6.0_45
```
Comment (FelixGV) [inline run.bat]: Hi @mingqin1,

Thanks for this contribution. I don't know much about Windows development, so please forgive me in advance for the newbie question.

Is it possible in a Windows environment that these variables (JAVA_HOME, MAVEN_HOME, etc.) be already set? And if yes, would there be a way for this script to honor the already-set variables instead of clobbering them? Given that these are hard-coded paths, and that they look non-standard, I would imagine it to be better not to make assumptions on where Java or Maven is installed.

What do you think?

Thanks.

-F

**Comment by mingqin1** [comment]
Felix:
 "Is it possible in a Windows environment that these variables (JAVA_HOME, MAVEN_HOME, etc.) be already set?"
 In window, there are environment variables of System Properties could be used to JAVA_HOME, MAVEN_HOME. see below link.  
   http://www.computerhope.com/issues/ch000549.htm

  Once JAVA_HOME ,MAVEN_HOME variables are set up as window environment variables,  window command prompt brought up window terminal will honor them. 

  Unlike Linux command   "alternatives --install  /user/bin/java java " to link java to multiple jdk version,  window has no  similar mechanism. 

   In window command prompt terminals, to alter jdk honored by JAVA_HOME in environment variable which always point to sole version jdk,  ones have to use run " set JAVA_HOME= c:\<specific jdk folder>" to change to desired jdk version  which is similar  to linux export

  In conclusion,  run.bat  could  not include JAVA_HOME, MAVEN_HOME variable settings, as long as accompanied document of running run.bat should mention jdk version and let users to determine how to  incorporate jre 1.6 for run server.

Ming

**Comment by eric239** [comment]
@mingqin1 what you stated is almost correct (you actually can have equivalent of links on windows, they're called "junctions" though people rarely use / are aware of them).

But the point @FelixGV made, and I want to reiterate, is that no files that we have in the shared repo should have anything hardcoded to a particular user's environment. Some options have reasonable non user-specific defaults such as `set MAVEN_OPTS=-Xms128m -Xmx512m` but using your local custom paths as you have with java and maven homes is not appropriate.

**Comment by mingqin1** [comment]
Ernest :
  I am fine on getting rid of local paths to JAVA_HOME or MAVEN_HOME in run.bat.
Thanks for pointing out  window "Junctions". 
Ming

**Comment by FelixGV** [comment]
Yeah, that's what I meant: we should remove any mention of a user's specific environment.

MAVEN_HOME and MAVEN_OPTS should be unnecessary in the context of the run script, right?

In the case of JAVA_HOME, if it is necessary in order for the run script to work, then can we instead have a check that that verifies if it is set, and bail out if it is not? Rather than making assumptions on where Java is installed on the user's machine?

Thanks (:

**Comment by FelixGV** [comment]
Can we finish cleaning this up so it contains no hard-coded user-specific paths? I don't use Windows so I can't test, and I don't really feel like setting up a VM just for that...

Thanks guys (: ...

**Comment by mingqin1** [comment]
Yes. I can clean it up

On Thu, Jan 29, 2015 at 9:00 AM, Felix GV <notifications@github.com> wrote:

> Can we finish cleaning this up so it contains no hard-coded user-specific
> paths? I don't use Windows so I can't test, and I don't really feel like
> setting up a VM just for that...
>
> Thanks guys (: ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/schema-repo/schema-repo/pull/33#issuecomment-72061604>
> .
>



-- 
Ming Qin
949-388-9898

**Comment by FelixGV** [comment]
Thanks @mingqin1,

I advise creating a branch for this, and rebasing off of the current schema-repo/master. Modifying your own master and sending a PR from it can cause extra work for you in the future, as the schema-repo/master evolves independently of yours, as is the case now.

Cheers!

-F

**Comment by mingqin1** [comment]
Felix:
  Can anyone have creating  branch access privilege of the repository to
create a new branch derived from the latest master?

On Thu, Feb 5, 2015 at 10:14 AM, Felix GV <notifications@github.com> wrote:

> Thanks @mingqin1 <https://github.com/mingqin1>,
>
> I advise creating a branch for this, and rebasing off of the current
> schema-repo/master. Modifying your own master and sending a PR from it can
> cause extra work for you in the future, as the schema-repo/master evolves
> independently of yours, as is the case now.
>
> Cheers!
>
> -F
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/schema-repo/schema-repo/pull/33#issuecomment-73096646>
> .
>



-- 
Ming Qin
949-388-9898

**Comment by FelixGV** [comment]
Hi @mingqin1,

I meant that you should do that in your own repo. There should be no need for you to create branches on the main repo in order to prepare properly rebased pull requests.

Let me know if that's unclear, and I'll provide git commands if necessary...

Thanks.

-F

**Comment by mingqin1** [comment]
Felix:
  A new branch forked from the  2/5//2014 master (
https://github.com/schema-repo/schema-repo  )  is listed as below.  This
branch has run.bat file which doesn't have jdk, maven environment variables
.

https://github.com/mingqin1/schema-repo/tree/run_batch_branch

On Thu, Feb 5, 2015 at 11:37 AM, Felix GV <notifications@github.com> wrote:

> Hi @mingqin1 <https://github.com/mingqin1>,
>
> I meant that you should do that in your own repo. There should be no need
> for you to create branches on the main repo in order to prepare properly
> rebased pull requests.
>
> Let me know if that's unclear, and I'll provide git commands if
> necessary...
>
> Thanks.
>
> -F
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/schema-repo/schema-repo/pull/33#issuecomment-73111450>
> .
>



-- 
Ming Qin
949-388-9898

**Comment by FelixGV** [comment]
Great, thanks a lot @mingqin1!

@eric239, would you mind reviewing Ming's latest batch file, since you have the most experience with Windows?

Please merge in or ask for corrections as you see fit.

Thanks guys!

-F

**Comment by eric239** [comment]
Will do later tonight

**Comment by eric239** [comment]
@mingqin1 
Stuff like `bundle/target/schema-repo-bundle-*-withdeps.jar` certainly won't work on windows (at least within the regular CMD.exe) - slashes need to be reversed and there is no bash-style globbing.

I'm promptly getting error trying to run (after fresh build)
```
M:\Work\Source\GD\workspace\schema-repo-bat>run file-system

Starting Schema Repo Server with local file system backend
Error: Could not find or load main class org.schemarepo.server.RepositoryServer
```

**Comment by mingqin1** [comment]
Ernest:
  I did "mvn install" , then "run file-system", the server is up.

On Thu, Feb 5, 2015 at 8:27 PM, Ernest Mishkin <notifications@github.com>
wrote:

> @mingqin1 <https://github.com/mingqin1>
> Stuff like bundle/target/schema-repo-bundle-*-withdeps.jar certainly
> won't work on windows (at least within the regular CMD.exe) - slashes need
> to be reversed and there is no bash-style globbing.
>
> I'm promptly getting error trying to run (after fresh build)
>
> M:\Work\Source\GD\workspace\schema-repo-bat>run file-system
>
> Starting Schema Repo Server with local file system backend
> Error: Could not find or load main class org.schemarepo.server.RepositoryServer
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/schema-repo/schema-repo/pull/33#issuecomment-73181217>
> .
>



-- 
Ming Qin
949-388-9898

**Comment by eric239** [comment]
Perhaps you're using cygwin?

**Comment by mingqin1** [comment]
My environment is window .

On Sun, Feb 8, 2015 at 5:57 PM, Ernest Mishkin <notifications@github.com>
wrote:

> Perhaps you're using cygwin?
>
> On Sun, Feb 8, 2015 at 4:29 PM, Ming Qin <notifications@github.com> wrote:
>
> > Ernest:
> > I did "mvn install" , then "run file-system", the server is up.
> >
> > On Thu, Feb 5, 2015 at 8:27 PM, Ernest Mishkin <notifications@github.com
> >
> > wrote:
> >
> > > @mingqin1 <https://github.com/mingqin1>
> > > Stuff like bundle/target/schema-repo-bundle-*-withdeps.jar certainly
> > > won't work on windows (at least within the regular CMD.exe) - slashes
> > need
> > > to be reversed and there is no bash-style globbing.
> > >
> > > I'm promptly getting error trying to run (after fresh build)
> > >
> > > M:\Work\Source\GD\workspace\schema-repo-bat>run file-system
> > >
> > > Starting Schema Repo Server with local file system backend
> > > Error: Could not find or load main class
> > org.schemarepo.server.RepositoryServer
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> > https://github.com/schema-repo/schema-repo/pull/33#issuecomment-73181217
> >
> > > .
> > >
> >
> >
> >
> > --
> > Ming Qin
> > 949-388-9898
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/schema-repo/schema-repo/pull/33#issuecomment-73442578>
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/schema-repo/schema-repo/pull/33#issuecomment-73447251>
> .
>



-- 
Ming Qin
949-388-9898
